trunk/SUAVE/Methods/Aerodynamics/Supersonic_Zero/Drag/parasite_drag_wing.py

```diff
@@ -0,0 +1,272 @@
+    Tc  = freestream.temperature    
+    re  = freestream.reynolds_number     
+    
+    wing = geometry
+    wing_parasite_drag = 0.0
+    
+    # Unpack wing
+    exposed_root_chord_offset = wing.exposed_root_chord_offset
+    symm                      = wing.symmetric
+    semispan                  = wing.spans.projected*0.5 * (2 - symm)
+    t_c_w                     = wing.thickness_to_chord
+    Sref                      = wing.areas.reference
+    num_segments              = len(wing.Segments.keys())     
+    
+    # if wing has segments, compute and sum parasite drag of each segment
+    
+    if num_segments>0:        
+        total_wetted_area            = 0
+        total_segment_parasite_drag  = 0 
+        total_segment_k_w            = 0 
+        total_segment_cf_w_u         = 0
+        total_segment_cf_w_l         = 0 
+        total_segment_k_comp_u       = 0
+        total_segment_k_comp_l       = 0
+        total_k_reyn_u               = 0  
+        total_k_reyn_l               = 0    
+        root_chord                   = wing.chords.root  
+        
+        for i_segs in range(num_segments):
+            if i_segs == num_segments-1:
+                continue 
+            else:  
+                span_seg  = semispan*(wing.Segments[i_segs+1].percent_span_location - wing.Segments[i_segs].percent_span_location )            
+                sweep_seg = wing.Segments[i_segs].sweeps.quarter_chord    
+                xtu       = wing.transition_x_upper
+                xtl       = wing.transition_x_lower      
+                
+                if i_segs == 0:
+                    chord_root    = root_chord*wing.Segments[i_segs].root_chord_percent
+                    chord_tip     = root_chord*wing.Segments[i_segs+1].root_chord_percent   
+                    wing_root     = chord_root + exposed_root_chord_offset*((chord_tip - chord_root)/span_seg)
+                    taper         = chord_tip/wing_root  
+                    mac_seg       = wing_root  * 2/3 * (( 1 + taper  + taper**2 )/( 1 + taper))  
+                    Sref_seg      = span_seg*(chord_root+chord_tip)*0.5 
+                    S_exposed_seg = (span_seg-exposed_root_chord_offset)*(wing_root+chord_tip)*0.5                    
+                
+                else: 
+                    chord_root    = root_chord*wing.Segments[i_segs].root_chord_percent
+                    chord_tip     = root_chord*wing.Segments[i_segs+1].root_chord_percent
+                    taper         = chord_tip/chord_root   
+                    mac_seg       = chord_root * 2/3 * (( 1 + taper  + taper**2 )/( 1 + taper))
+                    Sref_seg      = span_seg*(chord_root+chord_tip)*0.5
+                    S_exposed_seg = Sref_seg
+ 
+                if wing.symmetric:
+                    Sref_seg = Sref_seg*2
+                    S_exposed_seg = S_exposed_seg*2
+                
+                # compute wetted area of segment
+                if t_c_w < 0.05:
```

Comment:
these are really sizing things, maybe we should put these elsewhere. Because basically at every iteration you're doing basic sizing operations which could be done once.